convert2rhel/unit_tests/README.md

```diff
@@ -0,0 +1,170 @@
+After finishing, the command will create some hidden files in the root
+directory, such as `.install`, `.pre-commit`, `.images` and so on... Don't
+worry, this is just a way to tell `make` to not run the same commands over and
+over if you try.
+
+Optionaly, you can run the command below to assert that those hidden files were
+created succesfully.
+
+```bash
+# (Optional) Grep for all the hidden files in the directory
+ls -A | grep "^\."
+```
+
+## Writing unit tests
+
+Unit tests basically consist of testing small units of code, either a function
+call or a class definition to assert the code is doing what is expected.
+Usually, unit tests tend to mock all the external dependencies and other
+function calls that are not part of the testing code, but this is not something
+entirely necessary for every case. Mocking of external dependencies should be
+done sparingly for things that are unreliable (e.g. mocking a filepath, API
+calls, or some system-related commands).
+
+For instance, take the following test pseudo-code as an example of something
+that needs to be mocked:
+
+```python
+import mock
+import package_handler
+
+def test_check_for_yum_updates(monkeypatch):
+    packages_to_update_mock = mock.Mock(return_value=["package-1", "package-2"])
+    monkeypatch.setattr(package_handler, "get_packages_to_update", value=packages_to_update_mock)
+
+    assert package_handler.get_packages_to_update() is not None
+    assert packages_to_update_mock.assert_called_once_with(["package-1", "package-2"])
+```
+
+And this other example, of a test that don't need any mocks for external
+dependencies:
+
+```python
+import os
+import file_handler
+
+def test_archive_old_files(tmpdir):
+    tmpdir = str(tmpdir)
+    some_dir = os.path.join(tmpdir, "some_dir")
+    some_file = os.path.join(tmpdir, "some_file.txt")
+    test_data = "test data\n"
+
+    open(some_file, "w").write(test_data)
+
+    assert "some_file.txt" in os.listdir(tmpdir)
+
+    file_handler.archive_old_files(tmpdir)
+
+    old_files = os.listdir(some_dir)
+    assert len(old_files) == 1
+    with open(os.path.join(some_dir, old_files[0])) as old_f:
+        assert old_f.read() == test_data
+```
+
+The difference between the two is that, the first one we can't rely on the
+output of the `package_handler` (Because in every test run, the result may be
+different depending on the system, because we are relying on an external
+dependency), so we mock it to be sure we have full control over the results,
+and for the second one, we know it's such a simple function that just move old
+files to another directory, so it's better to provide the files and some tmp
+path, still having full control of the output, no mocking necessary.
+
+## Running the tests
+
+The tests can be run in two different ways, `Locally` or with `Docker`.
+
+Running the tests locally has the advantage of using a debugger to follow the
+execution of the code, and it's easier to generate coverage reports from the
+tests.
```

Comment:
Code coverage can be retrieved from the `make coverage`. If we want to we can also extend that command to generate a report.  
```suggestion
Running the tests locally has the advantage of using a debugger to follow the
execution of the code.
```

Reply:
Uh, we don't have `coverage` in our `Makefile` right now. But sure, we can do a make command just for that, I think it could be easier for people to remember, after all, running `make coverage` is more probably to be remembered than the whole command sequence for that.